### DISCUSSION TITLE
Construction of GTOs from exponents and coefficients in PySCF

### DISCUSSION THREAD

**Discussion by neutronengas** [pyscf/pyscf#2683] [Q&A] [answered]
Hi everyone,

as part of a private project I am currently trying to reconstruct the gto.eval_ao function. That requires me to manually build the GTOs associated with the respective basis sets. When I construct the GTOs according the formula stated [here](https://en.wikipedia.org/wiki/Gaussian_orbital#Cartesian_coordinates), they don't Match the ones provided from PySCF. Is PySCF using another formula under the hood?

Thank you!

**Comment by Eclair-Wu** [comment]
Do you mean the GTOs in Cartesian coordinates do not match the ones provided from PySCF?

**Comment by sunqm** [comment, marked as answer]
Except s and p functions, the Cartesian GTO is not normalized in pyscf. It might be the source of difference. Do you find the difference is a ratio such as `sqrt(4*pi/5)` , `sqrt(4*pi/15)` , or other factors in https://en.wikipedia.org/wiki/Table_of_spherical_harmonics#Real_spherical_harmonics ? The `eval_gto` function does not include these factors. Actually, so are the integrals provided by the mol.intor('intxxx_cart') functions.

**Comment by verena-neufeld** [comment]
Hi @neutronengas , thanks for your question. Has it been answered?
